spec/core_extensions_spec.rb

```diff
@@ -182,6 +182,14 @@
         end
       end
 
+      context 'when currency argument is a hash' do
+        subject(:hash) { {cents: 100, currency: {id: 'USD'}} }
+
+        it 'converts Hash to Money using id from currency hash' do
+          expect(hash.to_money).to eq(Money.new(100, 'USD'))
```

Comment:
I think this test uses default currency instead of the specified one, because there's no `:iso_code` key.
Can you please choose a non-default currency to reveal the issue and fix it?